### PR TITLE
collect transit connections

### PR DESCRIPTION
**Pull request by kyorav**

### CODE REVIEW

File: pkg/ibm/datamodel/test/data/demo-with-instances-config.json [hunk 1]
```diff
@@ -3095,5 +3095,6 @@
         }
     ],
     "load_balancers": [],
+    "transit_connections": null,
```
Comment (zivnevo) [inline pkg/ibm/datamodel/test/data/demo-with-instances-config.json:3098]: Why is it `null` and not `[]`?

File: pkg/ibm/vpc_collection.go [hunk 2]
```diff
@@ -378,3 +379,22 @@ func getPolicyRules(vpcService *vpcv1.VpcV1, lbID, listenerID string,
...
+	APIFunc := func(pageSize int64, next *string) (*tgw.TransitConnectionCollection, any, error) {
+		return tgwService.ListConnections(&tgw.ListConnectionsOptions{Limit: &pageSize, Start: next})
+	}
+	getArray := func(collection *tgw.TransitConnectionCollection) []tgw.TransitConnection {
+		return collection.Connections
+	}
+
+	transitCons, err := iteratePagedAPI(APIFunc, getArray)
+	if err != nil {
+		return nil, fmt.Errorf("[getTransitConnections] error getting transit connections: #{err}")
```
Comment (zivnevo) [inline pkg/ibm/vpc_collection.go:393]:
```suggestion
		return nil, fmt.Errorf("[getTransitConnections] error getting transit connections: %w", err)
```

**Review by zivnevo — CHANGES_REQUESTED** [review]
Just these two nits

File: pkg/ibm/datamodel/test/data/demo-with-instances-config.json (hunk 1 above)
Comment (kyorav) [reply pkg/ibm/datamodel/test/data/demo-with-instances-config.json]: fixed

File: pkg/ibm/vpc_collection.go (hunk 2 above)
Comment (kyorav) [reply pkg/ibm/vpc_collection.go]: fixed

**Pull request merged**